examples/counter/src/lib.rs

```diff
@@ -0,0 +1,38 @@
+use rock_n_rollup::{
+    core::{Application, Runtime},
+    plugins::{
+        database::{Database, Json},
+        logger::Logger,
+    },
+    services::external::{External, FromExternal},
+};
+
+pub struct Decrement {}
+
+impl FromExternal for Decrement {
+    fn from_external(input: Vec<u8>) -> Result<Self, ()> {
+        match input.as_slice() {
+            [0x00] => Ok(Decrement {}),
+            _ => Err(()),
+        }
+    }
+}
+
+pub fn descrement<R: Runtime>(rt: &mut R, _: External<Decrement>) {
```

Comment:
you can replace the code of decrement by the code of counter (and keep the signature of decrement)

To make it clearer, you can rename decrement to "decrement_entrypoint" or "decrement_transition"

Basically a transition is a function that takes an input (from the delayed inbox), apply it, and returns.

Don't forget to register the transition in the `main` function.

Reply:
I rename it as this [commit](4b736d4ef97a9a154bff9ca59eca465d6e0a0833)